jquery.redirect.js

```diff
@@ -167,15 +171,15 @@ ShareAlike - If you remix, transform, or build upon the material, you must distr
     });
   };
 
-  var removeNulls = function (values) {
+  var removeNulls = function (values, isRemoveFieldBlank) {
     var propNames = Object.getOwnPropertyNames(values);
     for (var i = 0; i < propNames.length; i++) {
       var propName = propNames[i];
       if (values[propName] === null || values[propName] === undefined) {
         delete values[propName];
       } else if (typeof values[propName] === 'object') {
-        values[propName] = removeNulls(values[propName]);
-      } else if (values[propName].length < 1) {
+        values[propName] = removeNulls(values[propName], isRemoveFieldBlank);
+      } else if (values[propName].length < 1 && isRemoveFieldBlank) {
```

Comment:
The default behavior now is to keep the blank fields, so you should keep it to avoid breaking changes.
I'd probably change the flag and behavior.
Perhaps `shouldKeepBlankFields` is a good naming? 
Also need to change the if to `else if (!shouldKeepBlankFields && isRemoveFieldBlank)`.